resize.js

```diff
@@ -220,73 +220,72 @@ image.encodeAndUpload = function (fileName, fileType, resolutionX, resolutionY,
     res.writeHead(200, {'Content-Type': supportedFileType(fileType)});
 
-    gm(file)
-      .size(function (err, size) {
-		if (err) {
-			console.error(err);
-			return;
-		}
-        var originalRatio = size.width / size.height;
-        var newRatio = resolutionX / resolutionY;
-
-        var cropX = 0;
-        var cropY = 0;
-        var cropWidth = size.width;
-        var cropHeight = size.height;
-
-        if (fit === 'crop') {
-          if (originalRatio > newRatio) {
-            var resizeFactor = size.height / resolutionY;
-            cropWidth = size.width / resizeFactor;
-            cropHeight = resolutionY;
-            cropX = (cropWidth - resolutionX) / 2;
-          }
-          else {
-            var resizeFactor = size.width / resolutionX;
-            cropWidth = resolutionX;
-            cropHeight = size.height / resizeFactor;
-            cropY = (cropHeight - resolutionY) / 2;
-          }
-
-          this.options({imageMagick : true})
-            .autoOrient()
-            .resize(cropWidth, cropHeight)
-            .crop(resolutionX, resolutionY, cropX, cropY)
-            .stream(fileType, function (err, stdout, stderr) {
-              var r = stdout.pipe(res);
-              r.on('finish', function () {
-                // This is to close the result while a background job will continue to process
-                log('info', 'Finished sending a converted image');
-                res.end();
-              });
-            });
+    gm(file).size(function (err, size) {
+      if (err) {
+        console.error(err);
+        return;
+      }
+      var originalRatio = size.width / size.height;
+      var newRatio = resolutionX / resolutionY;
+
+      var resizeFactor;
+      var cropX = 0;
+      var cropY = 0;
+      var cropWidth = size.width;
+      var cropHeight = size.height;
+
+      if (fit === 'crop') {
+        if (originalRatio > newRatio) {
+          resizeFactor = size.height / resolutionY;
+          cropWidth = size.width / resizeFactor;
+          cropHeight = resolutionY;
+          cropX = (cropWidth - resolutionX) / 2;
         }
         else {
-          this.options({imageMagick : true})
-            .autoOrient()
-            .resize(resolutionX, resolutionY)
-            .stream(fileType, function (err, stdout, stderr) {
-              var r = stdout.pipe(res);
-              r.on('finish', function () {
-                // This is to close the result while a background job will continue to process
-                log('info', 'Finished sending a converted image');
-                res.end();
-              });
-            });
+          resizeFactor = size.width / resolutionX;
+          cropWidth = resolutionX;
+          cropHeight = size.height / resizeFactor;
+          cropY = (cropHeight - resolutionY) / 2;
         }
+      }
+
+      var workImageClient = gm(file)
+        .options({imageMagick: true})
+        .autoOrient();
+
+      if (resizeFactor) {
```

Comment:
Might be nicer to check if `cropX` etc is set (or put that in an object of its own).